src/miniflask/util.py

```diff
@@ -144,3 +145,91 @@ def __call__(self, parser, namespace, values, option_string=None, return_enum=Fa
             if return_enum:
                 return enum
         setattr(namespace, self.dest, enum)
+
+
+# ===== #
+# Units #
+# ===== #
+
+# base unit class
+# (used to save the unit endings and the converter functions)
+class Unit:  # pylint: disable=too-few-public-methods
+
+    def __init__(self, name, converter_fn, units):
+        self.name = name
+        self.units = {}
+        for keys in units:
+            k_uid = keys[0]
+            for k in keys:
+                if k in self.units:
+                    raise ValueError(f"All Unit-Synonymes must be unique, but {k} has been used already.")
+                self.units[k] = k_uid
+        self.converter = converter_fn
+
+    def __call__(self, value, unit):
+        if " in " in unit:
```

Comment:
` in ` can be a variable so that it can be set easier. Is a regex necessary, or is the str compare sufficient?